falcon/api.py

```diff
@@ -556,7 +558,15 @@ def handle(req, resp, ex, params):
 
         # Insert at the head of the list in case we get duplicate
         # adds (will cause the most recently added one to win).
-        self._error_handlers.insert(0, (exception, handler))
+        try:
+            exception_tuple = tuple(exception)
+        except TypeError:
+            exception_tuple = tuple([exception, ])
+
+        if all(issubclass(exc, BaseException) for exc in exception_tuple):
```

Comment:
I like the way we are checking all members of the given iterable here in advance :+1:  Could we add a test for the mixed case, for instance, when not all members of a tuple are exception classes?

Reply:
well, a mixed case would be a pathological situation where we should raise an error, right? Or am I missing something? IIUC `exception` _must_ be a subclass of `BaseException`, nothing else would be meaningful here (at least from the little I know about falcon!). Please do correct me if I'm wrong...

Reply:
Yes, exactly, we should raise an error exactly as your code does.
I.e. we want to write a test or two here to test even error handling / malformed input. You could use, for instance, [`pytest.raises`](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions) to assert that the correct type of error was raised.

Reply:
oh sure, got it. I will add those breifly.